### DISCUSSION TITLE
How to perform one round of inference before the actual training starts?

### DISCUSSION THREAD

**Discussion by avijit9** [PyTorchLightning/pytorch-lightning-bolts#710] [General]
Hi,

I would like to perform one full round of inference before the training starts on the training dataset. What is the easiest and optimal way to do that?

TIA.

**Comment by senarvi** [comment]
I think the way to do that is to [run Trainer.test()](https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html#testing) before you start training. I would actually like the possibility of performing validation also when training starts, so that when I'm doing transfer learning, I would see the initial state in the validation curves. However, I think you have to perform a separate inference round.
